lib/OpenQA/Scheduler/Model/Jobs.pm

```diff
@@ -42,6 +42,12 @@ sub schedule ($self, $allocated_workers = {}, $allocated_jobs = {}) {
     my $free_workers = determine_free_workers($self->shuffle_workers);
     my $worker_count = $schema->resultset('Workers')->count;
     my $free_worker_count = @$free_workers;
+    my $running = $schema->resultset('Jobs')->count({state => [OpenQA::Jobs::Constants::EXECUTION_STATES]});
+    my $limit = $self->{config}->{scheduler}->{max_running_jobs};
+    if (defined $limit and $running + 1 > $limit) {
+        $self->emit('conclude');
```

Comment:
I must say this will likely be to be very annoying anyways. Let's say I want to run some verification job (e.g. some job with a worker class to run on a very specific worker I know for sure nothing else runs on) and than notice that nothing happens. As a normal user I'd be quick to blame openQA and write a message in #eng-testing. I'm not sure whether a message about this in the scheduler log will be discoverable by users.

I'm also wondering how I myself would workaround being blocked for an unknown time because the verification jobs I need to run for a ticket like setting up new workers are not executed. Could we perhaps run jobs with a low enough priority value after all (e.g. all negative priority values)?

Reply:
I added a log_debug now